### PR TITLE
[refactor](status) Refactor status handling in agent task

### PR DESCRIPTION
**Pull request by platoneko**
# Proposed changes

Issue Number: #11874

## Problem summary

1. Refactor `TaggableLogger`
2. Refactor status handling in agent task:
    1. Unify log format in TaskWorkerPool
    2. Pass `Status` to the top caller, and replace some `OLAPInternalError` with more detailed error message `Status`
    3. Premature return with the opposite condition to reduce indention
## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [x] I don't know
4. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
5. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
6. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
7. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by jackwener — COMMENTED** [review]
Please rebase to fix the P0

File: be/src/common/logconfig.cpp [hunk 1]
```diff
@@ -25,8 +25,7 @@
 #include <mutex>
 
 #include "common/config.h"
-#include "gutil/stringprintf.h"
-#include "util/logging.h"
+#include "common/logging.h"
```
Comment (jackwener) [inline be/src/common/logconfig.cpp:28]: Maybe it's redundant?

**Review by jackwener — DISMISSED** [review]
LGTM.
except maybe add redundant include.

File: be/src/util/hash_util.hpp [hunk 2]
```diff
@@ -20,8 +20,9 @@
 
 #pragma once
 
+#include <functional>
```
[the remark this answers is not in the document]
Comment (platoneko) [reply be/src/util/hash_util.hpp:23]: this header is used to include std::hash template.

File: be/src/util/cpu_info.h [hunk 3]
```diff
@@ -20,12 +20,12 @@
 
 #pragma once
 
+#include <glog/logging.h>
```
[the remark this answers is not in the document]
Comment (platoneko) [reply be/src/util/cpu_info.h:23]: DCHECK use logging.h

File: be/src/common/logconfig.cpp (hunk 1 above)
Comment (platoneko) [reply be/src/common/logconfig.cpp:28]: this cpp file implement `init_glog` declared in `common/logging.h`

File: be/src/olap/task/engine_storage_migration_task.cpp [hunk 4]
```diff
@@ -341,19 +290,10 @@ void EngineStorageMigrationTask::_generate_new_header(
 
 Status EngineStorageMigrationTask::_copy_index_and_data_files(
         const string& full_path, const std::vector<RowsetSharedPtr>& consistent_rowsets) const {
-    Status status = Status::OK();
     for (const auto& rs : consistent_rowsets) {
-        status = rs->copy_files_to(full_path, rs->rowset_id());
-        if (!status.ok()) {
-            Status ret = FileUtils::remove_all(full_path);
```
Comment (yiguolei) [inline be/src/olap/task/engine_storage_migration_task.cpp:348]: This line is removed, then there exists garbage in this folder?

File: be/src/agent/task_worker_pool.cpp [hunk 5]
```diff
@@ -1259,11 +1217,6 @@ void TaskWorkerPool::_report_tablet_worker_thread_callback() {
             DorisMetrics::instance()->report_all_tablets_requests_skip->increment(1);
             continue;
         }
-        if (!build_all_report_tablets_info_status.ok()) {
-            LOG(WARNING) << "build all report tablets info failed. status: "
-                         << build_all_report_tablets_info_status;
```
Comment (yiguolei) [inline be/src/agent/task_worker_pool.cpp:1264]: If build failed, some logic in handle report maybe wrong, it think this logic should not be removed.

File: be/src/olap/task/engine_batch_load_task.cpp [hunk 6]
```diff
@@ -344,35 +307,26 @@ Status EngineBatchLoadTask::_delete_data(const TPushReq& request,
...
+                                                   tablet_info_vec);
     if (!res.ok()) {
-        LOG(WARNING) << "fail to push empty version for delete data. "
-                     << "res=" << res << "tablet=" << tablet->full_name();
         DorisMetrics::instance()->delete_requests_failed->increment(1);
-        return res;
     }
-
-    LOG(INFO) << "finish to process delete data. res=" << res;
     return res;
```
Comment (yiguolei) [inline be/src/olap/task/engine_batch_load_task.cpp:330]: All these tasks have a same problem. If the task is success, there is no log. So if a table lost some data or data is wrong, we could not find what happened on this tablet. I think we should print some success log to help us debug. Maybe could add a log in EngineTask.

File: be/src/olap/task/engine_clone_task.cpp [hunk 7]
```diff
@@ -96,176 +93,108 @@ Status EngineCloneTask::_do_clone() {
...
-                    drop_status.precise_code() != OLAP_ERR_TABLE_NOT_FOUND) {
-                    // just log
-                    LOG(WARNING) << "drop stale cloned table failed! tablet id: "
-                                 << _clone_req.tablet_id;
-                }
-            }
-            status = Status::InternalError("Failed to clone tablet");
-        } else {
-            LOG(INFO) << "clone get tablet info success. tablet_id:" << _clone_req.tablet_id
-                      << ", schema_hash:" << _clone_req.schema_hash << ", signature:" << _signature
```
Comment (yiguolei) [inline be/src/olap/task/engine_clone_task.cpp:263]: In previous logic, _tablet_infos is only set in this logic, not in 233 logic. But in your code, _tablet_infos is always set.

File: be/src/olap/task/engine_clone_task.cpp [hunk 8]
```diff
@@ -324,36 +251,24 @@ Status EngineCloneTask::_make_and_download_snapshots(DataDir& data_dir,
...
 
-        st = _download_files(&data_dir, remote_url_prefix, local_data_path);
-        if (!st.ok()) {
-            LOG(WARNING) << "fail to download and convert tablet, remote=" << remote_url_prefix
-                         << ", error=" << st.to_string();
-            status = Status::InternalError("Fail to download and convert tablet");
-            // when there is an error, keep this program executing to release snapshot
-        }
+        status = _download_files(&data_dir, remote_url_prefix, local_data_path);
+        // when there is an error, keep this program executing to release snapshot
```
Comment (yiguolei) [inline be/src/olap/task/engine_clone_task.cpp:255]: Losing log here, should print log when download files failed.

File: be/src/olap/task/engine_storage_migration_task.cpp (hunk 4 above)
Comment (platoneko) [reply be/src/olap/task/engine_storage_migration_task.cpp:348]: `EngineStorageMigrationTask::_migrate` has done this garbage collection.

File: be/src/agent/task_worker_pool.cpp (hunk 5 above)
Comment (platoneko) [reply be/src/agent/task_worker_pool.cpp:1264]: In fact, `build_all_report_tablets_info` always returns OK, I even consider let `build_all_report_tablets_info` return void.

File: be/src/olap/task/engine_batch_load_task.cpp (hunk 6 above)
Comment (platoneko) [reply be/src/olap/task/engine_batch_load_task.cpp:330]: Such log will be print in `_push_worker_thread_callback`:
```c++
        if (status.ok()) {
            LOG_INFO("successfully execute push task")
                    .tag("signature", agent_task_req.signature)
                    .tag("tablet_id", push_req.tablet_id)
                    .tag("push_type", push_req.push_type);
            ++_s_report_version;
            finish_task_request.__set_finish_tablet_infos(tablet_infos);
        } else {
            LOG_WARNING("failed to execute push task")
                    .tag("signature", agent_task_req.signature)
                    .tag("tablet_id", push_req.tablet_id)
                    .tag("push_type", push_req.push_type)
                    .error(status);
        }
```

File: be/src/olap/task/engine_clone_task.cpp (hunk 8 above)
Comment (platoneko) [reply be/src/olap/task/engine_clone_task.cpp:255]: done

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**